src/isomedia/avc_ext.c

```diff
@@ -2341,6 +2457,113 @@ GF_Err hvcc_Size(GF_Box *s)
 	}
 	return GF_OK;
 }
+
+GF_Box *av1c_New() {
+	GF_AV1ConfigurationBox *tmp = (GF_AV1ConfigurationBox *)gf_malloc(sizeof(GF_AV1ConfigurationBox));
+	if (tmp == NULL) return NULL;
+	memset(tmp, 0, sizeof(GF_AV1ConfigurationBox));
+	tmp->type = GF_ISOM_BOX_TYPE_AV1C;
+	return (GF_Box *)tmp;
+}
+
+void av1c_del(GF_Box *s) {
+	GF_AV1ConfigurationBox *ptr = (GF_AV1ConfigurationBox*)s;
+	if (ptr->config) gf_odf_av1_cfg_del(ptr->config);
+	gf_free(ptr);
+}
+
+Bool av1_is_obu_header(ObuType obu_type);
+GF_Err av1c_Read(GF_Box *s, GF_BitStream *bs) {
+	AV1State state;
+	u8 reserved;
+	u64 read = 0;
+	GF_AV1ConfigurationBox *ptr = (GF_AV1ConfigurationBox*)s;
+	
+	if (ptr->config) gf_odf_av1_cfg_del(ptr->config);
+	GF_SAFEALLOC(ptr->config, GF_AV1Config);
+	memset(&state, 0, sizeof(AV1State));
+	state.config = ptr->config;
+
+	reserved = gf_bs_read_int(bs, 3);
+	if (reserved != 0)
+		return GF_NOT_SUPPORTED;
+	ptr->config->initial_presentation_delay_present = gf_bs_read_int(bs, 1);
+	if (ptr->config->initial_presentation_delay_present) {
+		ptr->config->initial_presentation_delay_minus_one = gf_bs_read_int(bs, 4);
+	} else {
+		reserved = gf_bs_read_int(bs, 4);
+		ptr->config->initial_presentation_delay_minus_one = AV1_INITIAL_PRESENTATION_DELAY_MINUS_ONE_MAX;
+	}
+	read += 1;
+	ptr->config->obu_array = gf_list_new();
+
+	while (read < ptr->size && gf_bs_available(bs)) {
+		u64 pos, obu_size;
+		ObuType obu_type;
+		GF_AV1_OBUArrayEntry *a;
+		
+		pos = gf_bs_get_position(bs);
+		if (gf_media_aom_av1_parse_obu(bs, &obu_type, &obu_size, &state) != GF_OK) {
+			GF_LOG(GF_LOG_ERROR, GF_LOG_CONTAINER, ("[ISOBMFF] could not parse AV1 OBU at position "LLU". Leaving parsing.\n", pos));
+			break;
+		}
+		assert(obu_size == gf_bs_get_position(bs) - pos);
```

Comment:
I agree that this test is debatable. This is a cross-check that what we parse has the right length.

> The current AV1/ISOBMFF spec requires that the size field is present for OBUs in the config. 

Ok but if there are not present, should we skip the OBU or add the size field?

> Some OBUs may not be parseable if the obu_size is not given (e.g. private metadata). I don't see how the bistream position can be updated by gf_media_aom_av1_parse_obu in that case. This assert should be removed and replaced with the above.

It can't. In my opinion the AV1 spec should not allow that in any way.